### PR TITLE
Validate: add support for kubernetes deployment validation with `src validate kube` subcommand

### PR DESCRIPTION
**Pull request by jdpleiness**
This PR adds the `src validate kube` subcommand for validating Sourcegraph deployments on Kubernetes. 

Features include:
- pod validation 
- service validation
- PVC validation 
- network connection validation between Sourcegraph services

Validations can also include warnings for non-failure states that should be addressed.

Kubernetes cluster configuration is handled via standard kubeconfig file in users home directory or can be specified with `--kubeconfig` flag. Users can also specify a specific namespace via `--namespace` flag.

Screenshot 2023-01-20 at 11 23 56 AM
Screenshot 2023-01-20 at 11 25 44 AM
Screenshot 2023-01-20 at 11 51 04 AM

TODO:
- [x] Add `--quiet` flag to silence command line output in CI/CD pipelines
- [x] Update CHANGELOG.md

### Test plan
- Locally tested changes using latest Sourcegraph Helm install.
- Added unit tests in `internal/validate/kube/kube_test.go`.
- ran go test ./... - all tests passing
- ran staticcheck ./... - no issues found in new code

### CODE REVIEW

**Review by abeatrix — APPROVED** [review]
👍

File: internal/validate/kube/kube.go [hunk 1]
```diff
@@ -0,0 +1,392 @@
...
+// Pods will validate all pods in a given namespace.
+func Pods(ctx context.Context, config *Config) ([]validate.Result, error) {
+	pods, err := config.clientSet.CoreV1().Pods(config.namespace).List(ctx, metav1.ListOptions{})
+	if err != nil {
+		return nil, err
+	}
+
+	var results []validate.Result
+
+	//TODO make concurrent
```
Comment (beyang) [inline internal/validate/kube/kube.go:144]: Remove TODO in committed code
```suggestion
```

File: internal/validate/kube/kube.go [hunk 2]
```diff
@@ -0,0 +1,392 @@
...
+// Services will validate all  services in a given namespace.
+func Services(ctx context.Context, config *Config) ([]validate.Result, error) {
+	services, err := config.clientSet.CoreV1().Services(config.namespace).List(ctx, metav1.ListOptions{})
+	if err != nil {
+		return nil, err
+	}
+
+	var results []validate.Result
+
+	//TODO make concurrent
```
Comment (beyang) [inline internal/validate/kube/kube.go:225]:
```suggestion
```

File: internal/validate/kube/kube.go [hunk 3]
```diff
@@ -0,0 +1,392 @@
...
+// PVCs will validate all persistent volume claims on a given namespace
+func PVCs(ctx context.Context, config *Config) ([]validate.Result, error) {
+	pvcs, err := config.clientSet.CoreV1().PersistentVolumeClaims(config.namespace).List(ctx, metav1.ListOptions{})
+	if err != nil {
+		return nil, err
+	}
+
+	var results []validate.Result
+
+	//TODO make concurrent
```
Comment (beyang) [inline internal/validate/kube/kube.go:261]:
```suggestion
```

**Review by beyang — APPROVED** [review]

File: cmd/src/validate_install.go [hunk 4]
```diff
@@ -90,26 +87,26 @@ Environmental variables
...
-			validationSpec = validate.DefaultConfig()
+			validationSpec = install.DefaultConfig()
 		}
 
 		envGithubToken := os.Getenv("SRC_GITHUB_TOKEN")
 
 		// will work for now with only GitHub supported but will need to be revisited when other code hosts are supported
 		if envGithubToken == "" {
-			return errors.Newf("%s failed to read `SRC_GITHUB_TOKEN` environment variable", validateWarningEmoji)
+			return errors.Newf("%s  failed to read `SRC_GITHUB_TOKEN` environment variable", validate.WarningSign)
```
[the remark this answers is not in the document]
Comment (jdpleiness) [reply cmd/src/validate_install.go:104]: The extra space is needed to format the output properly for that specific emoji in the terminal it seems.

**Pull request merged**